packages/wdio-browserstack-service/src/launcher.ts

```diff
@@ -190,6 +191,30 @@ export default class BrowserstackLauncherService implements Services.ServiceInst
         } catch (err: unknown) {
             PercyLogger.error(`Error while setting best platform for Percy snapshot at worker start ${err}`)
         }
+
+        // Setting up healing for those sessions where we don't add the service version capability as it indicates that the session is not being run on BrowserStack
+        if (!shouldAddServiceVersion(this._config, this._options.testObservability, caps)) {
+            try {
+                if (caps.browserName) {
+                    caps = await AiHandler.setup(this._config, this.browserStackConfig, this._options, caps, false)
+                } else {  // setting up healing in case caps.xyz.capabilities.browserName where xyz can be anything:
+                    const hasBrowserName = (cap: Options.Testrunner) =>
+                        cap &&
+                        cap.capabilities &&
+                        (cap.capabilities as Capabilities.BrowserStackCapabilities).browserName
+
+                    const isValid = Object.values(caps).length > 0 && Object.values(caps).some(hasBrowserName)
+
+                    if (isValid) {
+                        caps = await AiHandler.setup(this._config, this.browserStackConfig, this._options, caps, true)
+                    }
+                }
+            } catch (err) {
+                if (this._options.selfHeal === true) {
+                    BStackLogger.debug(`Error while setting up Browserstack healing Extension ${err}. Disabling healing for this session.`)
```

Comment:
customer visible error?

Reply:
We are adding logs only in case the selfHeal flag is set to true by the user. We should not display logs unless user wants to check the healing flow.